### ISSUE TITLE
Provide live benchmarks comparing Hugo and competitors

### ISSUE THREAD

**Issue by vrmiguel** [gohugoio/hugoDocs#1772]
Hugo proudly presents itself with the slogan "The world’s fastest framework for building websites" in its website and in its main repository.

If that's the case, it'd be nice to have live benchmarks comparing Hugo and competitors in its CI pipeline, similar to what [Deno](https://deno.land/benchmarks) does, for example, openly comparing its performance with direct competitors such as Node and Bun.

However, Hugo's slogan does not seem to be factual taking in consideration the benchmarks found throughout the internet.

## [Benchmark 1](https://github.com/grego/ssg-bench) - Blades vs. Zola vs. Hugo

Intel Core i5-2400S @ 2.66GHz, 4 physical cores, HDD:
```
blades: 2.52743 ms (±.28058)
zola: 33.75000 ms (±.81700)
hugo: 42.43000 ms (±11.75521)
```

Intel Core i5-3230M CPU @ 2.60GHz, 2 physical cores, SSD:
```
blades: 2.86216 ms (±.28282)
zola: 29.13000 ms (±.90172)
hugo: 45.71000 ms (±3.11542)
```

Rockchip 3399, 2+4 physical cores, eMMC:
```
blades: 10.11564 ms (±1.26302)
zola: 242.10000 ms (±18.54804)
hugo: 223.52000 ms (±27.17921)
```

## [Benchmark 2](https://eklausmeier.goip.de/blog/2021/11-13-performance-comparison-saaze-vs-hugo-vs-zola/) - Saaze vs. Zola vs. Hugo

![image](https://user-images.githubusercontent.com/36349314/178404018-02537124-c59e-4acf-9642-bb11ea692383.png)
![image](https://user-images.githubusercontent.com/36349314/178404187-c022fa56-525f-4f6f-937a-69da98e33514.png)

## [Benchmark 3](https://hugo-vs-nift.gitlab.io/) - Hugo vs. Nift

Site Generator | Real | User | Sys
-- | -- | -- | --
Hugo v0.80.0 | 4.222s | 11.644s | 1.276s
Nift v2.3.10 | 1.107s | 1.846s | 1.982s

## [Benchmark 4](https://github.com/Elderjs/elderjs/discussions/166) - Hugo vs. ElderJs, Zola and many others

![image](https://user-images.githubusercontent.com/36349314/178404550-5cf3738a-5835-4f1e-a9cc-762ca54baae2.png)

![image](https://user-images.githubusercontent.com/36349314/178404631-600bb495-a213-4279-8ffd-2cd363c0ab27.png)

# Previous discussion

I did notice that this topic has been approached before in #1681 and [hugo #7050](https://github.com/gohugoio/hugo/issues/7050) but I found @jmooring's comment quite unprofessional:

[In the absence of overwhelming, statistically meaningful data to the contrary, we are not changing the tagline.](https://github.com/gohugoio/hugoDocs/issues/1681#issuecomment-1079727846)

That's not how any of this works. If Hugo claims to be the fastest, naturally the burden of proof is on the developers of Hugo, not on anybody else.

**Comment by Xunjin** [comment]
At least has been, somewhat, removed the burden of proof here. Also to point out that kinda of phrase is more a marketing-wise to attract people, which sometimes can be misleading. Maybe it was in the time which was written the slogan?

I hope this issue will not be viewed as some kind of an attack to @jmooring previous answer or to the project itself. Being open to criticism, in a constructive way, is generally good.

**Comment by bep** [comment]
I'm in the process of making Hugo even more effective than it is today. This is mostly about making it less memory hungry. If I wanted, I could spend that time on creating an extremely fast static site generator that did very little but writing files to disk using all the core of my MacBook so I could beat some artificial benchmark. But that is not what I enjoy spending my time on.

And the thing is, to measure speed you need to consider what it does, how effective it is (how much work is done per time unit), and in the tag line mentioned above also lives the snapiness of Hugo's development life cycle, the really fast server rebuilds, live reloading in the browser etc. 

All of this could probably be measured, ut it's extremely hard, and again, not something we're going to spend a very con strained development budget on.

But I appreciate your input and concerns.

**Comment by bep** [comment]
Just an added comment, I tested the first benchmark on my MacBook

```
./bench.sh                                           ✘ 1 master ✱
blades: 7.81750 ms (±2.69119)
hugo: 22.40000 ms (±1.42828)
```
(I haven't installed Zola, so I commented out that).

Hugo does better compared to some tests posted above, maybe it's my beefier MacBook, maybe it's the newer Hugo version (v0.101.0 used here). I had the quick look at the benchmark in question and:

* The Hugo site produces lots of more dirs/files, `26 directories, 76 files` vs the Blade's output of `19 directories, 50 files`, which makes the build setup very different.
* It's a very small site that possibly favours Rust's faster startup time and possibly gives Hugo a penalty with its fully navigable content tree.

But again, comparing apples and oranges isn't very fruitful.
